scipy/stats/_distn_infrastructure.py

```diff
@@ -3237,6 +3237,9 @@ def __init__(self, a=0, b=inf, name=None, badvalue=None,
             raise ValueError("xk and pk need to have the same length.")
         if not np.allclose(np.sum(pk), 1):
             raise ValueError("The sum of provided pk is not 1.")
+        if np.any(np.asarray(xk).astype(int) != xk):
+            raise ValueError("xk must be integers. If you really want floats, "
+                             "use xk as integer indices into the float array.")
```

Comment:
Doing only that will only give you access to a proper subset of the functionality of the module; how, for instance, would you reproduce

	In [2]: from scipy.stats import rv_discrete

	In [4]: import numpy as np

	In [5]: xk = np.array([1.0, 1.5])

	In [6]: pk = np.array([0.5, 0.5])

	In [7]: rv = rv_discrete(values=(xk,pk))

	In [8]: rv.mean()
	Out[8]: 1.25

	In [9]: rv.var()
	Out[9]: 0.0625